domain/src/main/java/com/teamagam/gimelgimel/domain/location/DisplayUsersLocationInteractor.java

```diff
@@ -43,20 +45,39 @@ protected DisplayUsersLocationInteractor(ThreadExecutor threadExecutor,
     }
 
     @Override
-    protected Observable<UserEntity> buildUseCaseObservable() {
-        return createIntervalObservable()
-                .startWith((long) -1)
-                .flatMap(n -> mUsersLocationRepository.getLastLocations())
-                .mergeWith(mUsersLocationRepository.getUsersLocationUpdates())
-                .map(this::createUserEntity)
-                .doOnNext(mGeoEntitiesRepository::update)
-                .filter(ue->!mDisplayedEntitiesRepository.isShown(ue))
-                .doOnNext(mDisplayedEntitiesRepository::show);
+    protected Iterable<SubscriptionRequest> buildSubscriptionRequests(
+            DataSubscriptionRequest.SubscriptionRequestFactory factory) {
+        return Collections.singleton(buildDisplayRequest(factory));
+    }
+
+    private DataSubscriptionRequest<?> buildDisplayRequest(DataSubscriptionRequest.SubscriptionRequestFactory factory) {
+        return factory.create(
+                createIntervalObservable(),
+                observable -> observable
+                        .flatMap(n -> mUsersLocationRepository.getLastLocations())
+                        .mergeWith(mUsersLocationRepository.getUsersLocationUpdates())
```

Comment:
While you're at it... Since last locations is a "snapshot" of the current state, it doesn't need to return Observable of UserLocation but an Iterable of UserLocation.